smdebug/core/hook.py

```diff
@@ -794,6 +794,17 @@ def __init__(
         self.exported_collections = False
         self.data_type_name = data_type_name
 
+    def _is_suppported_dist_strategy(self):
+        num_workers = self._get_num_workers()
+        if num_workers > 1:
+            return True
+        else:
+            if self.first_process is None:
+                self.first_process = is_first_process(self.out_dir)
+                if self.first_process is False:
+                    self.save_all_workers = False
```

Comment:
can you warn the user about not saving from all workers? probably add a test case too - when the hook has include_workers="all" and the training script uses an unsupported distributed approach